server/services/assessmentService.ts

```diff
@@ -27,4 +31,32 @@ export default class AssessmentService {
 
     return assessment
   }
+
+  async initializePage(Page: TasklistPageInterface, request: Request, userInput?: Record<string, unknown>) {
+    const assessment = await this.findAssessment(request.user.token, request.params.id)
+    const body = getBody(Page, assessment, request, userInput)
+
+    return new Page(body, assessment, request.session.previousPage)
+  }
+
+  async save(page: TasklistPage, request: Request) {
+    const errors = page.errors()
+
+    if (Object.keys(errors).length) {
+      throw new ValidationError<typeof page>(errors)
+    } else {
+      const assessment = await this.findAssessment(request.user.token, request.params.id)
+
+      const pageName = getPageName(page.constructor)
+      const taskName = getTaskName(page.constructor)
+
+      assessment.data = assessment.application.data || {}
```

Comment:
Should this be `assessment.data`? At the moment we're passing the body of the whole application to the assessment, which I don't think we want